src/main/kotlin/com/openlattice/shuttle/hazelcast/serializers/IntegrationStreamSerializer.kt

```diff
@@ -0,0 +1,88 @@
+package com.openlattice.shuttle.hazelcast.serializers
+
+import com.hazelcast.nio.ObjectDataInput
+import com.hazelcast.nio.ObjectDataOutput
+import com.openlattice.client.RetrofitFactory
+import com.openlattice.data.integration.StorageDestination
+import com.openlattice.hazelcast.StreamSerializerTypeIds
+import com.openlattice.hazelcast.serializers.OptionalStreamSerializers
+import com.openlattice.hazelcast.serializers.TestableSelfRegisteringStreamSerializer
+import com.openlattice.hazelcast.serializers.UUIDStreamSerializer
+import com.openlattice.shuttle.control.Integration
+import org.springframework.stereotype.Component
+
+@Component
+class IntegrationStreamSerializer : TestableSelfRegisteringStreamSerializer<Integration> {
+
+    companion object {
+        private val environments = RetrofitFactory.Environment.values()
+        private val storageDestinations = StorageDestination.values()
+
+        fun serialize(output: ObjectDataOutput, obj: Integration) {
+            UUIDStreamSerializer.serialize(output, obj.key)
+            output.writeInt(obj.environment.ordinal)
+            output.writeInt(obj.defaultStorage.ordinal)
```

Comment:
This is totally cool, though I also have a base class for writing Enum serializers if they're used a lot (like *A LOT*, likely not the case here). Check out `AbstractEnumSerializer` sometime